Dream-no24/PlayerEnemyShipVariety
# Fix point count bug

## What
This PR fixes bug that enemy ships destroyed by explosive enemy doesn't give points

## Why
The points only add up when enemy ship is alive and ship is not hit by bullet(s).
An explosive enemy just destroys ships, so it can't add points because exploding ships are not hit by bullet(s).

## Related Issue
Closes #14 
Closes #16 : Fixed by limiting the explosion range of the explosive enemy ship

## How to Test
1. Run the game.
2. Hit an explosive enemy with player's bullet and check points add up correctly.
3. Check again in other levels.

## Additional Information
- Fixed some minor bugs caused in this branch.
  1. Fixed game freezing bug after destroyed all ships. (84c62b7)
  2. Fixed bug that explosion sprite disappears too fast during chain explosion. (3dae4c2)

- _destroy(Bullet, EnemyShip)  is changed to _destroy(Bullet, EnemyShip, boolean) in EnemyShipFormaintion.java
